src/plugins/upf/upf_pfcp_api.c

```diff
@@ -1065,9 +860,21 @@ handle_update_pdr (upf_session_t * sx, pfcp_update_pdr_t * update_pdr,
 	ASSERT (!pool_is_free_index (gtm->upf_apps, p[0]));
 	app = pool_elt_at_index (gtm->upf_apps, p[0]);
 	update->pdi.adr.application_id = p[0];
-	update->pdi.adr.db_id = upf_adf_get_adr_db (p[0]);
+	update->pdi.adr.db_id = upf_adf_get_adr_db (p[0], &acl);
 	update->pdi.adr.flags = app->flags;
 
+        if (!ISSET_BIT (pdr->pdi.grp.fields, PDI_SDF_FILTER))
+          {
+            vec_reset_length (update->pdi.acl);
```

Comment:
I don't think that this is valid. A PDI can contain SDFs **and** an Application Id. The ACL need to be merged.

Reply:
it's `!ISSET_BIT` not `ISSET_BIT`, so this reset only happens when there's **no** SDF filter present, only app id. If **both** app id and SDF are present, the vector is not reset.